### PR TITLE
[BEAM-11338] Addendum: handle private fields in generated thrift code

### PR DESCRIPTION
**Pull request by ccciudatu**
For thrift classes generated with `java:beans` or `java:private-members` (i.e. private fields), the current code will fail to infer the `FieldValueTypeInformation`, as it wrongly expects the fields to be public.
This PR fixes the issue and changes the thrift compiler params to showcase this scenario.

### CODE REVIEW

**Comment by ccciudatu** [comment]
R: @TheNeuralBit Just realised that https://github.com/apache/beam/pull/13428 does not work with the `beans` thrift compiler option, which renders all fields private. The fix is to use `getDeclaredField()` instead of `getField()`, thus making no assumptions about fields visibility.

File: sdks/java/io/thrift/src/test/java/org/apache/beam/sdk/io/thrift/ThriftIOTest.java [hunk 1]
```diff
@@ -72,16 +72,16 @@ public void setUp() throws Exception {
...
+    TEST_THRIFT_STRUCT.setTestByte((byte) 100);
+    TEST_THRIFT_STRUCT.setTestShort((short) 200);
+    TEST_THRIFT_STRUCT.setTestInt(2500);
+    TEST_THRIFT_STRUCT.setTestLong(79303L);
+    TEST_THRIFT_STRUCT.setTestDouble(25.007);
+    TEST_THRIFT_STRUCT.setTestBool(true);
+    TEST_THRIFT_STRUCT.setStringIntMap(new HashMap<>());
+    TEST_THRIFT_STRUCT.getStringIntMap().put("first", (short) 1);
+    TEST_THRIFT_STRUCT.getStringIntMap().put("second", (short) 2);
+    TEST_THRIFT_STRUCT.setTestBinary(buffer);
```
Comment (ccciudatu) [inline sdks/java/io/thrift/src/test/java/org/apache/beam/sdk/io/thrift/ThriftIOTest.java:84]: The older `ThriftIO` tests relied on fields being public.

File: sdks/java/io/thrift/src/test/java/org/apache/beam/sdk/io/thrift/ThriftIOTest.java [hunk 2]
```diff
@@ -240,15 +240,15 @@ public void testThriftCoder() {
...
-      temp.testLong = RandomUtils.nextLong();
+      temp.setStringIntMap(new HashMap<>());
+      temp.getStringIntMap().put(randomString, s);
+      temp.setTestShort(s);
+      temp.setTestBinary(buffer);
+      temp.setTestBool(RandomUtils.nextBoolean());
+      temp.setTestByte((byte) RandomUtils.nextInt(0, Byte.MAX_VALUE + 1));
+      temp.setTestDouble(RandomUtils.nextDouble());
+      temp.setTestInt(RandomUtils.nextInt());
+      temp.setTestLong(RandomUtils.nextLong());
```
Comment (ccciudatu) [inline sdks/java/io/thrift/src/test/java/org/apache/beam/sdk/io/thrift/ThriftIOTest.java:251]: The older `ThriftIO` tests relied on fields being public.

**Comment by ccciudatu** [comment]
R: @chrlarsen

**Comment by ccciudatu** [comment]
I can revert all test changes if they're too noisy, as the fixed schema now works with both beans and no-beans generated code (and with any possible combination of thrift compiler options, as I think I tried them all already).

File: sdks/java/io/thrift/src/test/java/org/apache/beam/sdk/io/thrift/ThriftIOTest.java [hunk 3]
```diff
@@ -240,15 +240,15 @@ public void testThriftCoder() {
...
       short s = (short) RandomUtils.nextInt(0, Short.MAX_VALUE + 1);
-      temp.stringIntMap = new HashMap<>();
-      temp.stringIntMap.put(randomString, s);
-      temp.testShort = s;
-      temp.testBinary = buffer;
-      temp.testBool = RandomUtils.nextBoolean();
-      temp.testByte = (byte) RandomUtils.nextInt(0, Byte.MAX_VALUE + 1);
-      temp.testDouble = RandomUtils.nextDouble();
-      temp.testInt = RandomUtils.nextInt();
-      temp.testLong = RandomUtils.nextLong();
```
Comment (ccciudatu) [inline sdks/java/io/thrift/src/test/java/org/apache/beam/sdk/io/thrift/ThriftIOTest.java:251]: The ThriftIO tests used to rely on fields being public.

**Review by TheNeuralBit — APPROVED** [review]
LGTM, thanks for catching this

**Pull request merged**